src/battle_script_commands.c

```diff
@@ -3801,6 +3801,53 @@ void SetMoveEffect(bool32 primary, u32 certain)
                     gBattlescriptCurrInstr = BattleScript_SpikesActivates;
                 }
                 break;
+            case MOVE_EFFECT_TRIPLE_ARROWS:
+                {
+                    u8 randomLowerDefenseChance;
+                    u8 randomFlinchChance;
+
+                    if (GetBattlerAbility(gBattlerAttacker) == ABILITY_SERENE_GRACE)
```

Comment:
Create a common util function to be used here and in the other location there's a Serene Grace check. That way only one function needs to be modified when the Rainbow is added eventually.